src/lightning_app/core/app.py

```diff
@@ -520,6 +526,84 @@ def state_dict(self) -> Dict:
     def load_state_dict(self, state: Dict) -> None:
         self.set_state(state)
 
+    @classmethod
+    def _load_checkpoint_from_json_file(cls, path: str) -> None:
+        checkpoint_dict = json.load(open(path))
+        # TODO: Checkpoint versioning and compatibility checks
+        return checkpoint_dict
+
+    def _get_checkpoint_if_available_locally(self, checkpoint: str) -> Optional[dict]:
+        checkpoint_dict = None
+        if os.path.exists(checkpoint):
+            checkpoint_dict = self._load_checkpoint_from_json_file(checkpoint)
+        if os.path.exists(os.path.join(self.checkpoint_dir, checkpoint)):
```

Comment:
I don't understand why there are two possible ways to load here. Isn't checkpoint a path?

Reply:
I want to allow either using the full path or only the name of the checkpoint.